liquidator/liquidator.js

```diff
@@ -158,12 +168,27 @@ class Liquidator {
       // Note: query the time again during each iteration to ensure the deadline is set reasonably.
       const currentBlockTime = this.empClient.getLastUpdateTime();
 
+      let tokensToLiquidate;
+      if (tokenBalanceWei) {
+        // Calculate the maximum tokens we can liquidate. We cannot bring the position down below the `minSponsorPosition`,
+        // and we cannot liquidate more than the bot's current balance, `tokenBalanceWei`. If the liquidator has enough balance
+        // to liquidate the entire position, then it will do so. If not, then it will liquidate the maximum amount of tokens it can provided
+        // that it maintains the position's token debt above the minimum token size.
+        const positionTokensAboveMinimum = this.toBN(position.numTokens).sub(this.toBN(this.empMinSponsorSize));
+        const maxTokensToLiquidate = this.BN.min(this.toBN(tokenBalanceWei), this.toBN(position.numTokens));
+        tokensToLiquidate = maxTokensToLiquidate.eq(this.toBN(position.numTokens))
+          ? maxTokensToLiquidate
+          : this.BN.min(positionTokensAboveMinimum, maxTokensToLiquidate);
```

Comment:
Took me quite a while to understand what's going on here (just because of the weird way the min position size affects liquidations). I wish this min position math was more intuitive :/.

Reply:
Yeah I agree its opaque. I am thinking of changing the variable name from `tokenBalanceWei` to `maxTokensToLiquidate` to make it more clear that this is the maximum amount of tokens that you will liquidate. It doesn't neccessarily have to tie to your balance.

I am also thinking of other ways to name `positionTokensAboveMinimum` but I'm not sure there's another way to make that calculation any more readable

Reply:
Yeah, tbh, I don't think this is an implementation issue. It's just that the math is weird and discontinuous.